partner-chains-cli/src/smart_contracts.rs

```diff
@@ -24,3 +25,65 @@ pub fn runtime_config_arguments(
 			payment_signing_key_path
 	)
 }
+
+pub fn check_for_kupo_ogmios_connection_error(
+	response: &str,
+	sidechain_main_cli_resources: &SidechainMainCliResources,
+) -> anyhow::Result<()> {
+	if response.contains("ECONNREFUSED") {
+		let kupo_port_str = &sidechain_main_cli_resources.kupo.port.to_string();
+		let (target, connection) = if response.contains(kupo_port_str) {
```

Comment:
Edge-case when it doesn't work: both services are at same port but different hosts.

Reply:
You're right. Unfortunately, there's nothing we can do, since the host printed by the CLI is not necessarily the same as the user provided (it looks like it resolves the host name to IP).

Reply:
We've decided to scrap this check alltogether